torch/_dynamo/output_graph.py

```diff
@@ -2323,6 +2323,12 @@ def __init__(self, output_graph, parent=None, is_export=False, source_target=Non
         # Only safe if we know for sure that *NOT* replaying these side-effects during
         # backward recomputation of the checkpoint region doesn't affect its correctness.
         self.allow_side_effects_under_checkpoint = False
+        # True if we want to allow side-effects (doesn't throw error on their existence)
+        # during this tracer's tracing. Unlike the above flag, this is not specific
+        # to torch.utils.checkpoint. You can flip this on by calling your function through
+        # `torch._dynamo.utils._UNSAFE_allow_side_effects(fn, *args, **kwargs)`. Note: Side-effects
+        # are allowed if this flag OR the above flag is True.
+        self.allow_side_effects = False
```

Comment:
Btw, it's not that we don't allow side effects, we don't allow side effects that are externally visible.

Maybe name this
self.unsafe_allow_externally_visible_side_effects ?